src/matrix/sync.rs

```diff
@@ -12,33 +12,43 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use futures::stream::Stream;
-use futures::{Async, Future, Poll};
+//use futures::stream::Stream;
+//use futures::{Async, Future, Poll};
 
 use serde_json;
 
 use std::io;
 
 use super::protocol::SyncResponse;
 
-use tokio_core::reactor::Handle;
-
+use std::pin::Pin;
 use url::Url;
 
-use crate::http::{HttpClient, HttpResponseFuture, Request};
+type CompatResponseFut = hyper::client::ResponseFuture;
+
+use futures3::prelude::{Future, Stream, TryFuture};
+use futures3::task::Poll;
+use std::task::Context;
 
-use slog::*;
+use crate::http;
 
 pub struct MatrixSyncClient {
     url: Url,
     access_token: String,
     next_token: Option<String>,
-    http_stream: HttpClient,
-    current_sync: Option<HttpResponseFuture>,
+    http_client: http::ClientWrapper,
+    current_sync: RequestStatus,
+}
+
+use std::boxed::Box;
```

Comment:
Why is this import all the way down here?

Reply:
@VanillaBrooks could you move this to the other imports?